controller/server/main.go

```diff
@@ -122,61 +116,60 @@ func newDeployment(req *cpb.CreateClusterRequest) (*deploy.Deployment, error) {
 }
 
 func (s *server) CreateCluster(ctx context.Context, req *cpb.CreateClusterRequest) (*cpb.CreateClusterResponse, error) {
-	log.Infof("Received CreateCluster request: %+v", req)
+	log.Infof("Received CreateCluster request: %v", req)
 	d, err := newDeployment(req)
 	if err != nil {
 		return nil, status.Errorf(codes.InvalidArgument, "unable to parse request: %v", err)
 	}
 	log.Infof("Parsed request into deployment: %v", d)
 	if err := d.Deploy(ctx, defaultKubeCfg); err != nil {
 		resp := &cpb.CreateClusterResponse{
-			Name:  req.GetKind().Name,
+			Name:  d.Cluster.GetName(),
 			State: cpb.ClusterState_CLUSTER_STATE_ERROR,
 		}
 		return resp, status.Errorf(codes.Internal, "failed to deploy cluster: %v", err)
 	}
-	log.Infof("Cluster %q deployed and ready for topology", req.GetKind().Name)
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.deployments[d.Cluster.GetName()] = d
```

Comment:
Do we need to check whether the cluster exists already?

Reply:
Good catch